newIDE/app/src/Profile/EditProfileDialog.js

```diff
@@ -57,6 +72,115 @@ export const getUsernameErrorText = (error: ?AuthError) => {
+                  ) {
+                    showAlert({
+                      title: t`You're awesome!`,
+                      message: t`Thanks for starring GDevelop repository. We added credits to your account as a thank you gift.`,
+                    });
+                  } else if (
+                    response.code === 'github-star/repository-not-starred'
+                  ) {
+                    showAlert({
+                      title: t`We could not find your GitHub star`,
+                      message: t`Make sure you star the repository called 4ian/GDevelop with your GitHub user and try again.`,
+                    });
+                  } else if (response.code === 'github-star/user-not-found') {
+                    showAlert({
+                      title: t`We could not find your GitHub user and star`,
+                      message: t`Make sure you create your GitHub account, star the repository called 4ian/GDevelop, enter your username here and try again.`,
+                    });
+                  } else {
+                    throw new Error(
+                      `Error while updating the GitHub star: ${response.code}.`
+                    );
+                  }
+                } catch (error) {
+                  console.error('Error while updating GitHub star:', error);
+                  showAlert({
+                    title: t`Something went wrong`,
+                    message: t`Make sure you have a proper internet connection or try again later.`,
+                  });
+                }
+              }}
+              icon={<Coin fontSize="small" />}
+              label={
+                hasGithubStarBadge ? (
+                  <Trans>Credits given</Trans>
+                ) : (
+                  <Trans>Claim</Trans>
+                )
+              }
+              disabled={hasGithubStarBadge || disabled}
+              primary
+              style={style}
+            />
+          )}
+          renderTextField={() => (
+            <TextField
+              value={githubUsername}
+              floatingLabelText={<Trans>GitHub username</Trans>}
+              fullWidth
+              translatableHintText={t`Your GitHub username`}
+              onChange={(e, value) => {
+                onSetGithubUsername(value);
+              }}
+              disabled={disabled}
+              maxLength={githubUsernameConfig.maxLength}
+              helperMarkdownText={i18n._(
+                !hasGithubStarBadge
+                  ? t`[Star the GDevelop repository](https://github.com/4ian/GDevelop) and add your GitHub username here to get ${(githubStarAchievement &&
+                      githubStarAchievement.rewardValueInCredits) ||
+                      '-'} free credits as a thank you!`
+                  : t`This will help matching your open-source contributions to your profile. Credits were added to your account as a thank you.`
```

Comment:
The future is confusing to me here, wdyt about `Thank you for supporting GDevelop open-source community. Credits were added to your account as a thank you.`
The mention of us matching them through their username could be a bit scary